libavpipe/src/avpipe_xc.c

```diff
@@ -1407,7 +1407,7 @@ prepare_audio_encoder(
         ecodec = params->ecodec2;
         encoder_context->audio_last_dts[i] = AV_NOPTS_VALUE;
 
-        encoder_context->audio_stream_index[output_stream_index] = output_stream_index;
+        encoder_context->audio_stream_index[i] = output_stream_index;
```

Comment:
This looks like a bug.  There are not many places in the code that use the encoder audio_stream_index array (the most important one is the flush code)